src/Camera.tsx

```diff
@@ -175,13 +212,24 @@ export class Camera extends React.PureComponent<CameraProps> {
    * ```
    */
   public startRecording(options: RecordVideoOptions): void {
+    this.setState(({ timestamps: prevTimestamps }) => ({
+      timestamps: {
+        ...prevTimestamps,
+        requestRecordingStartedAt: new Date().valueOf(),
+      },
+    }));
     const { onRecordingError, onRecordingFinished, ...passThroughOptions } = options;
     if (typeof onRecordingError !== 'function' || typeof onRecordingFinished !== 'function')
       throw new CameraRuntimeError('parameter/invalid-parameter', 'The onRecordingError or onRecordingFinished functions were not set!');
 
     const onRecordCallback = (video?: VideoFile, error?: CameraCaptureError): void => {
       if (error != null) return onRecordingError(error);
-      if (video != null) return onRecordingFinished(video);
+      if (video != null) {
+        return onRecordingFinished({
+          ...this.state.timestamps,
+          ...video,
+        });
+      }
```

Comment:
I think this will cause us problems getting it merged into the library itself.  Need a very good justification as it will be a breaking change.

Reply:
marking vision-camera library re-merged to original repository is hard.